torch/_inductor/template_heuristics.py

```diff
@@ -0,0 +1,600 @@
+            {"config": (16, 32, 32, 6, 2), "cond": True},
+            {"config": (16, 64, 32, 6, 2), "cond": True},
+            {"config": (16, 128, 32, 6, 4), "cond": True},
+            {"config": (16, 256, 32, 6, 4), "cond": True},
+            {"config": (16, 32, 64, 6, 2), "cond": True},
+            {"config": (16, 64, 64, 6, 2), "cond": True},
+            {"config": (16, 128, 64, 6, 4), "cond": True},
+            {"config": (16, 256, 64, 6, 4), "cond": True},
+            {"config": (32, 32, 32, 6, 2), "cond": True},
+            {"config": (32, 64, 32, 6, 2), "cond": True},
+            {"config": (32, 128, 32, 6, 4), "cond": True},
+            {"config": (32, 256, 32, 6, 4), "cond": True},
+            {"config": (32, 32, 64, 6, 2), "cond": True},
+            {"config": (32, 64, 64, 6, 2), "cond": True},
+            {"config": (32, 128, 64, 6, 4), "cond": True},
+            {"config": (32, 256, 64, 6, 4), "cond": True},
+        ]
+
+        self.scaled_persistent_mm_configs = [
+            # "BLOCK_M", "BLOCK_N", "BLOCK_K", "num_stages", "num_warps"
+            {"config": (128, 128, 64, 3, 8), "cond": True},
+            {"config": (128, 128, 128, 3, 8), "cond": True},
+            {"config": (128, 128, 128, 4, 8), "cond": True},
+            {"config": (128, 128, 128, 4, 4), "cond": True},
+            {"config": (128, 128, 128, 3, 4), "cond": True},
+            {"config": (128, 128, 128, 5, 4), "cond": True},
+            {"config": (128, 128, 128, 5, 8), "cond": True},
+            {"config": (128, 128, 128, 6, 8), "cond": True},
+            {"config": (128, 128, 64, 4, 8), "cond": True},
+        ]
+
+        # TODO: Unify with other gemm patterns, mm_plus_mm currently follows
+        # slightly different pattern than rest
+        self.mm_plus_mm_configs = [
+            {"config": (64, 64, 32, 2, 4), "cond": True},
+            {"config": (64, 64, 32, 3, 8), "cond": True},
+            {"config": (64, 64, 32, 4, 16), "cond": True},
+            {"config": (64, 32, 32, 4, 8), "cond": True},
+            {"config": (32, 64, 32, 4, 8), "cond": True},
+            {"config": (128, 128, 32, 1, 8), "cond": True},
+            {"config": (64, 64, 64, 1, 8), "cond": True},
+            {"config": (32, 32, 128, 1, 8), "cond": True},
+            {"config": (64, 64, 16, 2, 4), "cond": True},
+            {"config": (32, 32, 16, 1, 2), "cond": True},
+        ]
+
+        self.conv_configs = [
+            # "BLOCK_M", "BLOCK_N", "BLOCK_K", "num_stages", "num_warps"
+            {"config": (64, 256, 16, 2, 4), "cond": True},
+            {"config": (256, 64, 16, 2, 4), "cond": True},
+            {"config": (1024, 16, 16, 1, 8), "cond": True},
+            {"config": (128, 128, 32, 2, 8), "cond": True},
+            {"config": (64, 64, 32, 2, 4), "cond": True},
+            {"config": (64, 256, 32, 2, 8), "cond": True},
+            {"config": (256, 64, 32, 2, 8), "cond": True},
+        ]
```

Comment:
This constructor will be super expensive, and we will pay this cost multiple times even though each time we only read a single value from it.  I'd suggest using a singleton pattern (one for each class) so you aren't paying the cost of the constructor over and over.

I suspect this is what is causing the compile time regression, since if we need one list if configs we will construct tons of them and discard all but one.

Reply:
Good idea.